DataObjects/DataObjects/GeantEventEntry.h

```diff
@@ -0,0 +1,23 @@
+#ifndef DATAOBJECTS_GEANTEVENTENTRY_H
+#define DATAOBJECTS_GEANTEVENTENTRY_H
+
+//Include files
+#include "GaudiKernel/DataObject.h"
+#include "G4Event.hh"
+
+class GAUDI_API GeantEventEntry : public DataObject {
+ public:
+  GeantEventEntry() : m_event(0){};
+   virtual ~GeantEventEntry() {if (m_event!=0) delete m_event;};
```

Comment:
Very true!

Thanks for catching,
  Benedikt

On 17 Mar 2015, at 14:45, Sebastien Binet <notifications@github.com<mailto:notifications@github.com>> wrote:


In DataObjects/DataObjects/GeantEventEntry.h<https://github.com/HEP-FCC/FCCSW/pull/25#discussion_r26572861>:

> @@ -0,0 +1,23 @@
> +#ifndef DATAOBJECTS_GEANTEVENTENTRY_H
> +#define DATAOBJECTS_GEANTEVENTENTRY_H
> +
> +//Include files
> +#include "GaudiKernel/DataObject.h"
> +#include "G4Event.hh"
> +
> +class GAUDI_API GeantEventEntry : public DataObject {
> + public:
> +  GeantEventEntry() : m_event(0){};
> +   virtual ~GeantEventEntry() {if (m_event!=0) delete m_event;};


this should read:

virtual ~GeantEventEntry() {
delete m_event;
m_event = nullptr;
};

otherwise, double-deletes will happen.
also, probably one should use a smart pointer of some kind very early on in FCC-SW.

—
Reply to this email directly or view it on GitHub<https://github.com/HEP-FCC/FCCSW/pull/25/files#r26572861>.